moto/iam/access_control.py

```diff
@@ -258,9 +263,41 @@ def check_action_permitted(self, resource: str) -> None:
             elif permission_result == PermissionResult.PERMITTED:
                 permitted = True
 
+        if self._is_assuming_role_operation(resource):
+            permitted = self._check_role_trust_relationship(resource)
```

Comment:
This should still take the result of the policies into account, right? So something like:

```suggestion
            permitted = permitted and self._check_role_trust_relationship(resource)
```

Reply:
Fixed